### PR TITLE
remove all CONFIG and all digest_config

### PR DESCRIPTION
**Pull request by cobordism**
This is a reissue of https://github.com/ManimCommunity/manim/pull/763

please excuse the duplicate but this allowed me to use github's branch comparison feature also for my branch.

This is joint work with leotrs and behackl. See PR 763 for more info.

### CODE REVIEW

**Comment by cobordism** [comment]
For the record, all tests pass...  but you have to take the tests from this branch and not from the master branch.

File: manim/mobject/geometry.py
Comment (kolibril13) [inline manim/mobject/geometry.py:401]:
```suggestion
            start_angle = 0,0,
            angle = TAU,
```

File: manim/mobject/geometry.py
Comment (kolibril13) [inline manim/mobject/geometry.py:108]: Do we want to move parameters like tip_length somewhere else?
With this defined here, also Mobjects like Dot() will have parameters tip_length and tip_style.
I guess that this does not bother too much, but it also does not make that much sense

Comment (behackl) [reply manim/mobject/geometry.py:108]: Currently, you can also change these `TipableVMobject`-specific properties in a CONFIG dict of `Dot`, so I don't think that anything changed practically. With this branch, you can just set it as a keyword; that is new.

(It should be said that all of this `TipableVMobject` stuff is really messy and should be refactored generously; we could, for example, let `Arc` and `Line` simply inherit from `VMobject`, and then have `Arrow` inherit from both `Line` and a new `Tipable` class which enables all of the business with the tips.

But this definitely is a discussion for another PR. 😄

Comment (kolibril13) [reply manim/mobject/geometry.py:108]: yep :)

**Comment by kolibril13** [comment]
Looks good to me!
I just tested some things with the attributes and I did not run into any errors.
And I still have some thoughts about the attributes that I left here: https://github.com/ManimCommunity/manim/issues/787
as this pr is maybe not the best place to discuss that.

File: manim/mobject/geometry.py
Comment (cobordism) [reply manim/mobject/geometry.py:108]: The idea here is to move everything into the __init__ function as keyword arguments and then later, clean each one up in turn.. 
So the answer is: we want to do that in a subsequent PR once CONFIGs are gone.

**Review by leotrs — APPROVED** [review]

**Comment by leotrs** [comment]
Let's merge this first thing after the release this week.

**Comment by kolibril13** [comment]
Amazing, even the docs are working!! :)
https://manimce--783.org.readthedocs.build/en/783/examples.html

**Review by kolibril13 — APPROVED** [review]

**Review by naveen521kk — CHANGES_REQUESTED** [review]
I think we shouldn't break things altogether. I would suggest to keep `digest_config` method and just warn users that using `CONFIG` is deprecated and would be removed in next release. And make things work with `CONFIG` atleast for a version.

**Comment by cobordism** [comment]
> I think we shouldn't break things altogether. I would suggest to keep `digest_config` method and just warn users that using `CONFIG` is deprecated and would be removed in next release. And make things work with `CONFIG` atleast for a version.

ugh!

the problem with digest_config is that is reads the CONFIGS from all classes in the hierarchy, resolves duplicates in its own way and sets attributes. It is _extremely_ unclear when and where which attributes are set and as such it makes is really hard to refactor and clean up individual classes. That is why we wanted to do it all at once. 

However, since 0.1.1 was just released, we have a whole release cycle to make this version nicer. Should you feel so inclined, you could for example write a new version of digest_config that read a config class variable for the local class in question and used those values as the new default values for the keyword arguments to `__init__`. 

I advocate for a clean break.

**Comment by naveen521kk** [comment]
Maybe let's hold this PR for the next release. First, make a small PR for making a deprecated warning whenever `digest_config` method is called so users can know that something crazy is going to happen soon and then it's ok to do this.

**Comment by cobordism** [comment]
Id' like to add for context:

- We have already held this back to come only after the 0.1.1 release but we did not include a deprecation warning in 0.1.1.

- Moving from 0.1.1 -> 0.2.0 can be a breaking change. When the leading digit is 0, semantic versioning allows breaking changes when the second digit (minor version) is increased. Meanwhile 1.0.0 suggests a major release and I think that is very premature.

**Comment by naveen521kk** [comment]
> We have already held this back to come only after the 0.1.1 release but we did not include a deprecation warning in 0.1.1.

I don't think 🤔 we added any warning about that? If we had then it could be merged or else IMHO it should wait for another release.

**Comment by cobordism** [comment]
> wait for another release.

I worry that it will take quite an effort maintaining the PR as merge ready. Almost every other PR affects this one in some way requiring manual intervention to merge / rebase and even entries in the documentation have to be rewritten (as indeed you can see in the contents of this PR as they are now). I do not relish this task. 

Why do you think is so important to announce in 0.1.1 the changes that will be in 0.2.0? Can you elaborate perhaps? Will it not be enough to prepare good documentation along with the next release? ..bearing in mind also that 0.1.1 remains available in future to compile any user animations.

**Comment by leotrs** [comment]
@naveen521kk I agree this is a major breaking change. I also think it should be merged ASAP. A number of things are being held up because of the existence of CONFIG. (And, as @cobordism said, it was basically bypassing python's inheritance mechanisms.) 

Perhaps a good compromise is to prepare a good documentation page titled "migrating your classes away from CONFIG" that we can release together with our next release.

**Comment by eulertour** [comment]
I also think this needs to be merged now for all the reasons that have been brought up before. There isn't a good way to allow for both of them when the CONFIG system is so odd.

**Review by naveen521kk — APPROVED** [review]
Ok then LGTM!

**Comment by cobordism** [comment]
@leotrs will you do the honours?

**Comment by naveen521kk** [comment]
Let me do.

**Comment by kolibril13** [comment]
🎉

File: tests/test_renderer.py
Comment (behackl) [inline tests/test_renderer.py:1]: Have these tests moved somewhere else?

Comment (cobordism) [reply tests/test_renderer.py:1]: I only removed import statements when my linter told me that the import was unused.
If they were moved, I must assume they were moved prior to this PR

Comment (behackl) [reply tests/test_renderer.py:1]: It looks like the file `test_renderer.py` was removed (but is still present in `v0.1.1`)

Comment (cobordism) [reply tests/test_renderer.py:1]: hm... https://github.com/leotrs/manim/commit/7d83b902c8dae03c66591a3ebeadc529c459305e
@leotrs ?

Comment (leotrs) [reply tests/test_renderer.py:1]: That test was really only ever testing that the `video_quality_config` was behaving correctly, but this PR deleted video_quality_config so the test became obsolete.

**Pull request merged**
